bin/circuit_transpiler/src/main.cpp

```diff
@@ -0,0 +1,495 @@
+       << "                blob,\n"
+       << "                gate_params.offset\n"
+       << "            );\n"
+       << "        }\n"
+       << "        gate_params.selector_evaluations = new uint256[](GATES_N);\n"
+       << "        for (uint256 i = 0; i < GATES_N; i++) {\n"
+       << "            gate_params.selector_evaluations[i] = batched_lpc_verifier\n"
+       << "                .get_z_i_j_from_proof_be(\n"
+       << "                    blob,\n"
+       << "                    gate_params.eval_proof_selector_offset,\n"
+       << "                    i + ar_params.permutation_columns + ar_params.permutation_columns + "
+          "ar_params.constant_columns,\n"
+       << "                    0\n"
+       << "                );\n"
+       << "        }\n"
+       << "\n"
+       << "        uint256 t = 0;\n"
+       << "        assembly {\n"
+       << "            let modulus := mload(gate_params)\n"
+       << "            let theta_acc := 1\n"
+       << "            function get_eval_i_by_rotation_idx(idx, rot_idx, ptr) -> result {"
+       << "                result := calldataload(\n"
+       << "                    add(\n"
+       << "                        mload(add(add(ptr, 0x20), mul(0x20, idx))),\n"
+       << "                        mul(0x20, rot_idx)\n"
+       << "                    )\n"
+       << "                )\n"
+       << "            }\n"
+       << "\n"
+       << "            function get_selector_i(idx, ptr) -> result {\n"
+       << "                result := mload(add(add(ptr, 0x20), mul(0x20, idx)))\n"
+       << "            }\n"
+       << "            let x1 := add(gate_params, CONSTRAINT_EVAL_OFFSET)\n"
+       << "            let x2 := add(gate_params, WITNESS_EVALUATIONS_OFFSET)\n"
+       << "            let x3 := get_eval_i_by_rotation_idx(0,0,mload(x2))\n"
+       << "            let x4 := get_eval_i_by_rotation_idx(2,0,mload(x2))\n";
+
+    ProfilingType::process(os, constraint_system, columns_rotations);
+    os << "}\n}\n}" << std::endl;
+}
+
+template<typename ProfilingType, typename ConstraintSystemType, typename ColumnsRotationsType,
+         typename ArithmetizationParams>
+void print_sol_files(ConstraintSystemType &constraint_system, ColumnsRotationsType &columns_rotations,
+                     std::string out_folder_path = ".") {
+    ProfilingType::process_split(constraint_system, columns_rotations, out_folder_path);
+}
+
+template<typename FRIParamsType, typename TableDescriptionType, typename ColumnsRotationsType,
+         typename ArithmetizationParams>
+void print_params(FRIParamsType &fri_params, TableDescriptionType table_description,
+                  ColumnsRotationsType &columns_rotations, std::string output_folder) {
+    std::ofstream out;
+
+    out.open(output_folder + "/circuit_params.json");
+    out << "{\"arithmetization_params\":[" << ArithmetizationParams::witness_columns << ","
```

Comment:
Looks like you're building JSON with primitive tools. Isn't there any JSON library for converting data structures?